flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/AbstractTestBase.java

```diff
@@ -19,81 +19,61 @@
 package org.apache.flink.test.util;
 
 import org.apache.flink.configuration.Configuration;
-import org.apache.flink.runtime.akka.AkkaUtils;
-import org.apache.flink.runtime.minicluster.LocalFlinkMiniCluster;
 import org.apache.flink.util.FileUtils;
 
 import org.junit.ClassRule;
 import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.File;
 import java.io.IOException;
-import java.util.Objects;
-
-import scala.concurrent.duration.FiniteDuration;
 
 /**
- * A base class for tests that run test programs in a Flink mini cluster.
+ * Base class for unit tests that run multiple tests and want to reuse the same
+ * Flink cluster. This saves a significant amount of time, since the startup and
+ * shutdown of the Flink clusters (including actor systems, etc) usually dominates
+ * the execution of the actual tests.
+ *
+ * <p>To write a unit test against this test base, simply extend it and add
+ * one or more regular test methods and retrieve the StreamExecutionEnvironment from
+ * the context:
+ *
+ * <pre>
+ *   {@literal @}Test
+ *   public void someTest() {
+ *       ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+ *       // test code
+ *       env.execute();
+ *   }
+ *
+ *   {@literal @}Test
+ *   public void anotherTest() {
+ *       StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+ *       // test code
+ *       env.execute();
+ *   }
+ *
+ * </pre>
  */
 public abstract class AbstractTestBase extends TestBaseUtils {
 
-	/** Configuration to start the testing cluster with. */
-	protected final Configuration config;
+	protected static final Logger LOG = LoggerFactory.getLogger(AbstractTestBase.class);
 
-	private final FiniteDuration timeout;
+	private static final int DEFAULT_PARALLELISM = 4;
 
-	protected int taskManagerNumSlots = 1;
+	protected static final Configuration MINICLUSTER_CONFIGURATION = new Configuration();
 
-	protected int numTaskManagers = 1;
+	@ClassRule
```

Comment:
`miniClusterResource` will be initialized before `MINICLUSTER_CONFIGURATION` can be modified by a `@BeforeClass` method, i.e., for some cases the configuration cannot be supplied in time.

Reply:
Will remove the `MINICLUSTER_CONFIGURATION` completely.